DuckDuckGo/NetworkProtectionStatusView.swift

```diff
@@ -40,7 +40,7 @@ struct NetworkProtectionStatusView: View {
             toggle()
             locationDetails()
 
-            if statusModel.shouldShowConnectionDetails {
+            if statusModel.isNetPEnabled && statusModel.shouldShowConnectionDetails && statusModel.ipAddress != nil {
                 connectionDetails()
             }
```

Comment:
This should be changed before merging, the correct fix is to make sure that `shouldShowConnectionDetails` is accurate. I'm opening the PR now anyway since I need these changes for another branch, but will aim to update this by EOD.

Reply:
I posted this in MM but cross-posting here so it's easy to track the conversation.  I think the GUI logic is just wrong here, and I wouldn't change BSK.

Showing or hiding UI shouldn't rely directly on notifications, but on local state.  In this case it should rely on connection status.  This is the way macOS does it.

We don't ever want to show details when disconnected, and we always want to show details when connected.